src/components/common/Calculator.js

```diff
@@ -1,30 +1,126 @@
+import InfoIcon from '@material-ui/icons/Info';
 import _ from 'lodash';
+import Typography from '@material-ui/core/Typography';
 import Grid from '@material-ui/core/Grid';
 import { withTranslation } from 'react-i18next';
 import Result from './Result';
 import KeyPad from './Keypad';
 import { RESULT_ERROR_MESSAGE } from '../../constants/messages';
 import {
-  BUTTONS_NAMES,
+  BUTTONS,
   MAX_NUMBER_PRECISION,
+  BUTTONS_NAMES,
+  POWER_SYMBOL,
+  CALCULATOR_MAX_WIDTH,
+  ENABLED_COLOR,
+  DISABLED_COLOR,
   PI_SYMBOL,
   TIMES_SYMBOL,
 } from '../../constants/constants';
 
+const styles = () => ({
+  indicator: {
+    fontSize: '1rem',
+  },
+});
+
 class Calculator extends Component {
   static propTypes = {
     t: PropTypes.func.isRequired,
+    standalone: PropTypes.bool.isRequired,
+    classes: PropTypes.shape({
+      indicator: PropTypes.string.isRequired,
+    }).isRequired,
   };
 
   state = {
     mathjs: '',
     result: '',
+    isFocused: false,
   };
 
-  onClick = ({ name, text, katex, mathjs }) => {
+  componentDidMount() {
+    const { standalone } = this.props;
+
+    // handle keyboard input
+    if (standalone) {
+      this.setState({ isFocused: true });
+      window.addEventListener('keydown', this.handleKeydown);
+    }
+    // handle keyboard only when the iframe is focused
+    else {
+      window.addEventListener('focus', () => {
+        this.setState({ isFocused: true });
+        return window.addEventListener('keydown', this.handleKeydown);
+      });
+      window.addEventListener('blur', () => {
+        this.setState({ isFocused: false });
+        return window.removeEventListener('keydown', this.handleKeydown);
+      });
+    }
+  }
+
+  handleKeydown = (event) => {
+    const { key } = event;
+
+    // remove focus on previous clicked element
+    document.activeElement.blur();
+    let buttonName = null;
+    switch (key) {
+      case '0':
+      case '1':
+      case '2':
+      case '3':
+      case '4':
+      case '5':
+      case '6':
+      case '7':
+      case '8':
+      case '9':
+      case '+':
+      case '-':
+      case '.':
+        // find corresponding button object
+        buttonName = key;
+        break;
+      case 'Enter':
+      case '=':
+        buttonName = BUTTONS_NAMES.EQUAL;
+        break;
+      case 'Backspace':
+        buttonName = BUTTONS_NAMES.CE;
+        break;
+      case '*':
+        buttonName = BUTTONS_NAMES.MULTIPLY;
+        break;
+      case '/':
+        buttonName = BUTTONS_NAMES.DIVIDE;
+        break;
+      case POWER_SYMBOL:
+        buttonName = BUTTONS_NAMES.POWER;
+        break;
+      case '(':
+        buttonName = BUTTONS_NAMES.OPEN_PARENTHESIS;
+        break;
+      case ')':
+        buttonName = BUTTONS_NAMES.CLOSE_PARENTHESIS;
+        break;
+      default:
+        break;
+    }
+
+    if (buttonName) {
+      const button = BUTTONS.find(({ name }) => name === buttonName);
+      this.updateResult(button);
```

Comment:
`updateResult` takes an objet. Is `button` an object with those properties? It's not clear.

Reply:
Yes, the buttons are built using the buttons array in `src/constants.js`. So when the buttons are clicked, it triggers the event with the corresponding button properties. As for the keyboard, we can simply infer the button's name depending on the pressed key and retrieve its properties using its name.

Reply:
OK!